tests/vc_soa/vc_soa.cpp

```diff
@@ -0,0 +1,139 @@
+/** Algebra plugins library, part of the ACTS project
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.f
+ */
+
+// Project include(s).
+#include "algebra/vc_soa.hpp"
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+#include <array>
+#include <iostream>
+
+using namespace algebra;
+
+using scalar_t = float;
+
+constexpr float tol{1e-6f};
+
+/// This test the vector functions on an SoA (Vc::Vector) based vector
+TEST(test_vc_host, vc_soa_vector) {
```

Comment:
Is this test really for SoA (multiple vectors)? It looks like it is testing operations between two vectors

Reply:
Each object of type ```vc_soa::vector3<scalar_t>;``` contains a number of vectors equal to the simd width, if that is what you mean.

Reply:
So, ```a```, for example will look like this (float, avx2):
```
[1,1, 1, 1, 1, 1, 1, 1]
[2, 2, 2, 2, 2, 2, 2, 2]
[3, 3, 3, 3, 3, 3, 3, 3]
```
In an actual application, the numbers should not all be the same of course, but it was easier to write the test this way. There is, for example, no parametrized constructor for ```Vc::Vector```, if I remember correctly, because the number of elements it can hold depends on the floating point precision and the hardware capabilities

Reply:
Yes that's what I meant. I was just ignorant.
To make the tests more reliable, I would put different vectors for each simd element. But that's your choice

Reply:
Our comments get crossed.
 The number of SIMD elements cannot be accessed in run time?

Reply:
No, it depends on the floating point precision and on the particular CPU, because the number of bits in the vector registers is fixed. So, if I want to fill the a single simd element with different values, I have to make that dependent on these conditions as well

Reply:
Ah, wait accessed? It can, and has to be accessed at compile time, actually. I just cannot write
````
Vc::Vector<scalar> a{1, 2, 3, 4, 5, 6, 7, 8};
```
or rely on the fact that ```a[6]``` exists.

Reply:
I think it would be easy to do with a ```Vc``` inbuilt, though. I just did something similar in detray... Then we would be able to check that ```Vc``` implements its operations correctly, I guess

Reply:
Maybe it can be accessed with a predefined variable:
https://vcdevel.github.io/Vc-1.4/vcmacros.html#vc_size

But I don't know anything about it so you need to judge it

Reply:
Maybe you meant that it is difficult to decide whether to use double and float size, so yeah just forget about it if it is a hassle

Reply:
Thought there is a  `Vector::size()` that seems to be accessible in runtime

Reply:
Yes, this is even accessible at compile-time. But I still need a for loop or a lambda and a prescription on how to set the entries for every element of the soa vector and also for the results at every step to compare them against, right? Because, then I cannot compare against ```value_t(1.f)``` etc. anymore. I am not saying it is impossible, I am just not sure I have time for that now

Reply:
What I meant with the Vc inbuilt is ```Vc::Vector<scalar>::IndexesFromZero()``` which will give me reasonable starting vectors at least

Reply:
You don't have to do that if the time matters. 
Just FYI, you can put array with any pattern. For example, for n-th vector of `a` is set to [n, n+1, n+2]  and n-th vector of `b` to [n,n-1,n-2] and you can expect the output of addition, cross product, etc.

Reply:
Yes, this is pretty much what ```IndexesFromZero()``` allows you to do as a one-liner. I just don't have the time left to actually do it...

Reply:
Ah yes